x/qgb/orchestrator/querier.go

```diff
@@ -0,0 +1,269 @@
+	for _, val := range valset.Members {
+		vals[val.GetEthereumAddress()] = val
+	}
+
+	majThreshHold := valset.TwoThirdsThreshold()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, nil
+		case <-time.After(timeout):
+			return nil, fmt.Errorf("failure to query for majority validator set confirms: timout %s", timeout)
+		default:
+			currThreshHold := uint64(0)
+			confirmsResp, err := queryClient.DataCommitmentConfirmsByCommitment(
+				ctx,
+				&types.QueryDataCommitmentConfirmsByCommitmentRequest{
+					Commitment: commitment,
+				},
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, dataCommitmentConfirm := range confirmsResp.Confirms {
+				val, has := vals[dataCommitmentConfirm.EthAddress]
+				if !has {
+					return nil, fmt.Errorf(
+						"dataCommitmentConfirm signer not found in stored validator set: address %s nonce %d",
+						val.EthereumAddress,
+						valset.Nonce,
+					)
+				}
+				currThreshHold += val.Power
+			}
+
+			if currThreshHold >= majThreshHold {
+				return confirmsResp.Confirms, nil
+			}
+			q.logger.Debug(
+				"foundDataCommitmentConfirms",
+				fmt.Sprintf(
+					"total power %d number of confirms %d",
+					currThreshHold,
+					len(confirmsResp.Confirms),
+				),
+			)
+		}
+		// TODO: make the timeout configurable
+		time.Sleep(time.Second * 30)
+	}
+}
+
+func (q *querier) QueryTwoThirdsValsetConfirms(
+	ctx context.Context,
+	timeout time.Duration,
+	valset types.Valset,
+) ([]types.MsgValsetConfirm, error) {
+	// create a map to easily search for power
+	vals := make(map[string]types.BridgeValidator)
+	for _, val := range valset.Members {
+		vals[val.GetEthereumAddress()] = val
+	}
+
+	majThreshHold := valset.TwoThirdsThreshold()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, nil
+		// TODO: remove this extra case, and we can instead rely on the caller to pass a context with a timeout
+		case <-time.After(timeout):
+			return nil, fmt.Errorf("failure to query for majority validator set confirms: timout %s", timeout)
+		default:
+			currThreshHold := uint64(0)
+			queryClient := types.NewQueryClient(q.qgbRPC)
+			confirmsResp, err := queryClient.ValsetConfirmsByNonce(ctx, &types.QueryValsetConfirmsByNonceRequest{
+				Nonce: valset.Nonce,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			for _, valsetConfirm := range confirmsResp.Confirms {
+				val, has := vals[valsetConfirm.EthAddress]
+				if !has {
+					return nil, fmt.Errorf(
+						"valSetConfirm signer not found in stored validator set: address %s nonce %d",
+						val.EthereumAddress,
+						valset.Nonce,
+					)
+				}
+				currThreshHold += val.Power
+			}
+
+			if currThreshHold >= majThreshHold {
+				return confirmsResp.Confirms, nil
+			}
+			q.logger.Debug(
+				"foundValsetConfirms",
+				fmt.Sprintf(
+					"total power %d number of confirms %d",
+					currThreshHold,
+					len(confirmsResp.Confirms),
+				),
+			)
+		}
+		// TODO: make the timeout configurable
+		time.Sleep(time.Second * 30)
```

Comment:
I don't think these timeouts make sense here anymore

Reply:
But wouldn't that be too much ? If we keep sending requests non stop checking if we meet the threshold ? 
I guess we should create issues for this.